src/integrations/Http2Integration.ts

```diff
@@ -0,0 +1,291 @@
+import Integration from './Integration';
+import * as opentracing from 'opentracing';
+import {
+    Http2Tags,
+    SpanTags,
+    SpanTypes,
+    DomainNames,
+    ClassNames,
+    TriggerHeaderTags,
+} from '../Constants';
+import Utils from '../utils/Utils';
+import ModuleUtils from '../utils/ModuleUtils';
+import * as url from 'url';
+import ThundraLogger from '../ThundraLogger';
+import HttpError from '../error/HttpError';
+import ThundraSpan from '../opentracing/Span';
+import ThundraChaosError from '../error/ThundraChaosError';
+import ExecutionContextManager from '../context/ExecutionContextManager';
+
+const shimmer = require('shimmer');
+const has = require('lodash.has');
+const semver = require('semver');
+
+const thundraCollectorEndpointPattern1 = /^api[-\w]*\.thundra\.io$/;
+const thundraCollectorEndpointPattern2 = /^([\w-]+\.)?collector\.thundra\.io$/;
+
+const MODULE_NAME_HTTP2 = 'http2';
+
```

Comment:
/**
 * {@link Integration} implementation for HTTP2 integration
 * through {@code http2}  module
 */

Reply:
added jsdoc